src/hotspot/cpu/aarch64/frame_aarch64.inline.hpp

```diff
@@ -152,6 +152,7 @@ inline bool frame::is_older(intptr_t* id) const   { assert(this->id() != NULL &&
 
 inline intptr_t* frame::link() const              { return (intptr_t*) *(intptr_t **)addr_at(link_offset); }
 
+inline bool frame::can_access_link(Thread *thread) const { return thread->is_in_full_stack((address)addr_at(link_offset)); }
```

Comment:
is there a reason Thread* is non-const in all your variants of can_access_link and is_first_c_frame?

Reply:
No there is none.